scarb/src/ops/package.rs

```diff
@@ -486,3 +489,48 @@ fn tar(
     encoder.finish()?;
     Ok(uncompressed_size)
 }
+
+// Checks that the package has some piece of metadata that a human can
+// use to tell what the package is about.
+fn check_metadata(pkg: &Package, config: &Config) -> Result<()> {
+    let md = &pkg.manifest.metadata;
+    let mut missing = vec![];
+
+    trait IsEmpty {
+        fn is_empty(&self) -> bool;
+    }
+    impl IsEmpty for Utf8PathBuf {
+        fn is_empty(&self) -> bool {
+            self.to_string().is_empty()
+        }
+    }
+    macro_rules! lacking {
+        ($( $($field: ident)||* ),*) => {{
+            $(
+                if $(md.$field.as_ref().map_or(true, |s| s.is_empty()))&&* {
+                    $(missing.push(stringify!($field).replace("_", "-"));)*
+                    missing.push(String::from(" "));
+                }
+            )*
+        }}
+    }
+    lacking!(
+        readme,
+        description,
+        license || license_file,
+        documentation || homepage || repository
+    );
+    if !missing.is_empty() {
+        let messages = missing
+            .split(|elem| elem == " ")
+            .filter(|vec| !vec.is_empty())
+            .map(|vec| String::from("manifest has no ") + &*vec.join(" or "))
+            .collect::<Vec<String>>();
+        for message in messages {
+            config.ui().warn(message);
+        }
+        config.ui().print("see https://docs.swmansion.com/scarb/docs/reference/manifest.html#package for more info\n");
```

Comment:
```suggestion
        config.ui().print("see https://docs.swmansion.com/scarb/docs/reference/manifest.html#package for more info");
```
Unless you feel we need it

Reply:
Hmm, felt like newline makes message clearer - without it in the example the error and the warn seem to be a one message instead of two separate ones, wdyt?
```
warn: manifest has no description
warn: manifest has no license or license-file
warn: manifest has no documentation or homepage or repository
see https://docs.swmansion.com/scarb/docs/reference/manifest.html#package for more info

error: cannot archive package `workspaces` without a `lib` target
help: add `[lib]` section to package manifest
 --> Scarb.toml
+   [lib]
```

or 

```
warn: manifest has no description
warn: manifest has no license or license-file
warn: manifest has no documentation or homepage or repository
see https://docs.swmansion.com/scarb/docs/reference/manifest.html#package for more info
error: cannot archive package `workspaces` without a `lib` target
help: add `[lib]` section to package manifest
 --> Scarb.toml
+   [lib]
```